src/meshes/t8code_mesh.jl

```diff
@@ -100,6 +100,125 @@ function Base.show(io::IO, ::MIME"text/plain", mesh::T8codeMesh)
     end
 end
 
+"""
+    T8codeMesh{NDIMS, RealT}(forest, boundary_names; polydeg = 1, mapping=identity)
+
+Create a 'T8codeMesh'.
+
+# Arguments
+- `forest`: Pointer to a committed forest.
+- `boundary_names`: List of boundary names.
+- `polydeg::Integer`: Polynomial degree used to store the geometry of the mesh.
+                      The mapping will be approximated by an interpolation polynomial
+                      of the specified degree for each tree.
+- `mapping`: a function of `NDIMS` variables to describe the mapping that transforms
+             the imported mesh to the physical domain. Use `nothing` for the identity map.
+"""
+function T8codeMesh{NDIMS, RealT}(forest::Ptr{t8_forest}, boundary_names; polydeg = 1,
+                                  mapping = nothing) where {NDIMS, RealT}
+    # In t8code reference space is [0,1].
+    basis = LobattoLegendreBasis(RealT, polydeg)
+    nodes = 0.5 .* (basis.nodes .+ 1.0)
+
+    cmesh = t8_forest_get_cmesh(forest)
+    num_trees = t8_forest_get_num_global_trees(forest)
+
+    tree_node_coordinates = Array{RealT, NDIMS + 2}(undef, NDIMS,
+                                                    ntuple(_ -> length(nodes), NDIMS)...,
+                                                    num_trees)
+
+    coords_ref = Vector{Float64}(undef, 3)
+
+    # Calculate node coordinates of reference mesh.
+    if NDIMS == 2
+        num_corners = 4 # quadrilateral
+
+        # Testing for negative element volumes.
+        verts = zeros(3, num_corners)
+        for itree in 1:num_trees
+            veptr = t8_cmesh_get_tree_vertices(cmesh, itree - 1)
```

Comment:
I am a fan of more descriptive variable names such as `vertices` and `vertices_pointer` etc.... but I'll leave it up do you whethe ryou want to change it or not 😅

Reply:
Me too! Let's do it!